CMakeLists.txt

```diff
@@ -106,14 +110,85 @@ else()
   list(APPEND RAJAPERF_COMPILER_OPTIONS ${CMAKE_CXX_FLAGS})
 endif()
 
-configure_file(${CMAKE_SOURCE_DIR}/src/rajaperf_config.hpp.in
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/rajaperf_config.hpp.in
   ${CMAKE_CURRENT_BINARY_DIR}/bin/rajaperf_config.hpp)
 
 # Make sure RAJA flag propagate (we need to do some house cleaning to 
 # remove project-specific CMake variables that are no longer needed)
 set (CUDA_NVCC_FLAGS ${RAJA_NVCC_FLAGS})
 
+# The statement below is required for Kokkos compilation.
+if(ENABLE_KOKKOS)
+  include_directories(SYSTEM ${CMAKE_CURRENT_SOURCE_DIR}/tpl/RAJA/include/)
+endif()
+
+
+
+# ENABLE KOKKOS IS A RAJA PERFSUITE OPTION
+if(ENABLE_KOKKOS)
+  add_definitions(-DRUN_KOKKOS)
+  if(ENABLE_HIP)
+    set(Kokkos_ENABLE_HIP ON CACHE BOOL "Kokkos builds for AMD HIP set the
+Kokkos_ENABLE_HIP variable to ON")	
+    #set(Kokkos_ARCH_VEGA900 ON CACHE BOOL "Docstring") #TODO: better
+    #set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE /ascldap/users/ajpowel/RAJAPerf/amd_build/compiler_unscrewer)
+  endif()
+  if(ENABLE_TARGET_OPENMP)
+    set(Kokkos_ENABLE_OPENMPTARGET ON CACHE BOOL "Docstring")	
+    set(Kokkos_ARCH_VOLTA70 ON CACHE BOOL "Docstring") #TODO: better
+    set(CMAKE_CXX_STANDARD 17)
+    set(BLT_CXX_STANDARD 17)
+    set(RAJA_ENABLE_TARGET_OPENMP ON CACHE BOOL "Docstring")
+    if(NOT CMAKE_BUILD_TYPE MATCHES Debug)
+      if(NOT EXPERIMENTAL_BUILD)
+	      message(FATAL_ERROR "Kokkos builds with OpenMPTarget require a Debug build to succeed at the moment. Rebuild with CMAKE_BUILD_TYPE=Debug. If you're a compiler developer, rebuild with -DEXPERIMENTAL_BUILD=ON")
+      endif()
+    endif()
+
+
+
+
+
+
+
+    #add_definitions(-DRAJA_ENABLE_TARGET_OPENMP)
+  endif()
+
+# ENABLE_CUDA IS A RAJA PERFSUITE OPTION
+  if(ENABLE_CUDA)
+	  set(Kokkos_ENABLE_CUDA ON CACHE BOOL "Docstring")
+	  set(Kokkos_ENABLE_CUDA_LAMBDA ON CACHE BOOL "Docstring")
+	  set(Kokkos_ARCH_VOLTA70 ON CACHE BOOL "Docstring") #TODO: better
+    enable_language(CUDA)
+  endif()
+  if(ENABLE_OPENMP)
+    #set(Kokkos_ENABLE_OPENMP CACHE BOOL ON)
+    set(Kokkos_ENABLE_OPENMP ON CACHE BOOL "Docstring")
+  endif()
+
+  add_subdirectory(tpl/kokkos)
+  if(ENABLE_CUDA)
```

Comment:
This block is empty.

Reply:
I deleted the if (ENABLE_CUDA) and endif() bloc.